components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -353,6 +353,18 @@ var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
       newMoreInfoRow.insert(varMoreInfoInner);
       variantFooter.insert({before : newMoreInfoRow});
 
+      // restrict only numerical input for start_position and end_position variant fiellds
+      newMoreInfoRow.select('input:[id$=_start_position]', 'input:[id$=_end_position]').each( function(input) {
+        input.observe('keyup', function(event){
+          this.value=this.value.replace(/[^\d]/,'');
+        });
+      });
+
+      // get the last value of reference_genome of variants related to current gene symbol to create a new variant row with this value
```

Comment:
In that scenario, if user creates 2 genes and first adds a variant to the second gene, then starts adding variants to the first gene, then all variants will be created with the reference-genome value if the second gene, just bacause it us allways the last one, even if for tge first gene user wants different one.